### PR TITLE
Traverse StackTrace only once to enumerate it

### PR DESCRIPTION
**Pull request by diogokorok**

### CODE REVIEW

File: src/stacky_ffi.erl [hunk 1]
```diff
@@ -2,6 +2,14 @@
 
 -export([stacky_erlang_stack_trace/0]).
 
+enumerate_reversed(List) ->
+    I = length(List),
+    enumerate_reversed(I, 1, List).
+
+enumerate_reversed(I, S, List) ->
+    {List1, _ } = lists:mapfoldl(fun(T, Acc) -> {{Acc, T}, Acc-S} end, I, List),
+    List1.
```
Comment (diogokorok) [inline src/stacky_ffi.erl:11]: Implementation based on what's illustrated [in the docs](https://www.erlang.org/doc/apps/stdlib/lists.html#enumerate/3)

Comment (inoas) [reply src/stacky_ffi.erl:11]: in which docs?

Comment (inoas) [reply src/stacky_ffi.erl:11]: and... would you add a note to the changelog?

Comment (diogokorok) [reply src/stacky_ffi.erl:11]: > in which docs?

Oops, had an issue of malformatted markdown, I updated my comment to correctly link to it.

Comment (diogokorok) [reply src/stacky_ffi.erl:11]: > and... would you add a note to the changelog?

Sure

**Comment by inoas** [comment]
So the code base prior to your commit said that it was Apache 2 licensed but no license file was included. Now I have included one and switched it to MPL-2.0. Are you happy with that including your change? (MPL2 seems not to be an afferno style license, sadly, and it is not viral in so far that any chances not made to the files themselves do not need publication)

**Comment by inoas** [comment]
If you are unhappy, let me know and we will work it out. I just forgot to add a license really.

**Comment by inoas** [comment]
Thank you for this PR <3

**Pull request merged**